src/systems/sensors/Sensors.cc

```diff
@@ -521,6 +628,52 @@ Sensors::~Sensors()
   this->dataPtr->Stop();
 }
 
+//TODO: why does math::Vector3d work but not math::Vector3i?
+static void ConvertDoubleToUInt32x3(sdf::ElementConstPtr _parentElem,
+    const char* _childName, uint32_t _valueToSet[3], math::Vector3d _defaultValue)
+{
+  math::Vector3d parsedValues = (_parentElem == nullptr) ? 
+      _defaultValue : _parentElem->Get<math::Vector3d>(_childName, _defaultValue).first;
+
+  _valueToSet[0] = static_cast<uint32_t>(parsedValues[0]);
+  _valueToSet[1] = static_cast<uint32_t>(parsedValues[1]);
+  _valueToSet[2] = static_cast<uint32_t>(parsedValues[2]);
+}
+
+static void ConvertDoubleToUInt32x3(uint32_t _valueToSet[3], math::Vector3d _defaultValue)
+{
+  ConvertDoubleToUInt32x3(nullptr, "", _valueToSet, _defaultValue);
+}
```

Comment:
Here's some slight tweaks to these helper functions:

* consider renaming the functions to better distinguish between the two
* rearranged some logic in the functions - I think they still provide the same results.
* used lower case for static functions
* made `math::Vector3d` argument const ref

```suggestion
static void convertVector3dToUInt32Array(uint32_t _valueToSet[3],
    const math::Vector3d &_vecValues)
{
  _valueToSet[0] = static_cast<uint32_t>(_vecValues[0]);
  _valueToSet[1] = static_cast<uint32_t>(_vecValues[1]);
  _valueToSet[2] = static_cast<uint32_t>(_vecValues[2]);
}

static void parseVector3dAsUInt32Array(sdf::ElementConstPtr _parentElem,
    const char *_childName, uint32_t _valueToSet[3],
    const math::Vector3d &_defaultValue)
{
  math::Vector3d parsedValues = (_parentElem == nullptr) ? _defaultValue :
      _parentElem->Get<math::Vector3d>(_childName, _defaultValue).first;

  convertVector3dToUInt32Array(_valueToSet, parsedValues);
}
```

Reply:
can you also document the arguments?

Reply:
Args documented in f82c68e8b6a8e75ea5b01d2e168c7666f12b46f6